### ISSUE TITLE
API enhancement: set encodings by value

### ISSUE THREAD

**Issue by jakevdp** [altair-viz/altair#228]
Currently you can do, e.g.

``` python
Chart(data).mark_point().encode(
    x=X(value=5.0),
)
```

We could fairly easily allow this to be expressed

``` python
Chart(data).mark_point().encode(
    x=5.0,
)
```

It would involve modifying the `shorthand` trait in the template that generates `channel_collections.py`, and then modify `parse_shorthand` to appropriately handle non-string inputs. The `to_altair()` code would have to be modified as well to recognize the special case when only `value` is provided and also when the value is not a string.

**Comment by jakevdp** [comment]
The question is, is the additional complication in the code worth the simplification of the API?

**Comment by jakevdp** [comment]
Solved this in version 2.0 using ``x=alt.value(5.0)``
